Create script to compile sandboxes

pvollan opened the pull request:
#### 9816876160529fe04f0877a49bcd85b99ede78b3
<pre>
Create script to compile sandboxes
<a href="https://bugs.webkit.org/show_bug.cgi?id=264093">https://bugs.webkit.org/show_bug.cgi?id=264093</a>
<a href="https://rdar.apple.com/117854457">rdar://117854457</a>

Reviewed by Elliot Williams.

* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/compile-sandbox.sh: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/270714@main">https://commits.webkit.org/270714@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2bc85b2628ef037013eb3392f9b1a99647c18e07

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25877 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4485 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27155 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27974 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23692 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6248 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1913 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23778 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26126 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3381 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22291 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28554 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/26039 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2996 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29309 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23615 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23623 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27184 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3021 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1238 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4412 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6299 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3477 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3337 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Scripts/compile-sandbox.sh:
```diff
@@ -0,0 +1,37 @@
+SDK_NAME=$3
+SANDBOX_IMPORT_DIR=$4
+
+xcrun --sdk $SDK_NAME -f sbutil;
```
Before you land, let's silence the error from xcrun here.
```suggestion
xcrun --sdk $SDK_NAME -f sbutil 2>/dev/null;
```

emw-apple replied:
Oh, actually, we can't use xcrun like this at all — `set -e` means that a failure here aborts the script, which is presumably why it's failing on some builders.

How about testing it like this?
```
if xcrun --sdk $SDK_NAME -f sbutil 2> /dev/null; then
   # ...
fi
```

pvollan replied:
Ah, good point! I have fixed this in the latest patch.

Thanks for reviewing!

brentfulgham reviewed (approved):
That cleans things up nicely! r=me

pvollan committed:
Create script to compile sandboxes
https://bugs.webkit.org/show_bug.cgi?id=264093
rdar://117854457

Reviewed by Elliot Williams.

* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/compile-sandbox.sh: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/270714@main
